src/modules/units/components/CreateUnitsDialog.tsx

```
@@ -0,0 +1,162 @@
+  ]);
+
+  const handleSubmit = useCallback(() => {
+    if (!unitType || !numberOfUnits) return;
+
+    createUnits({
+      variables: {
+        input: {
+          input: {
+            projectId,
+            unitGroupId,
+            unitTypeId: unitType,
+            count: numberOfUnits,
+          },
+        },
+        includeCeFields: canViewCoordinatedEntry,
+      },
+    });
+  }, [
+    unitType,
+    numberOfUnits,
+    createUnits,
+    projectId,
+    unitGroupId,
+    canViewCoordinatedEntry,
+  ]);
+
+  const {
+    data: { pickList: unitTypePickList } = {},
+    loading: unitTypePickListLoading,
+    error: unitTypePickListError,
+  } = useGetPickListQuery({
+    variables: {
+      pickListType: PickListType.PossibleUnitTypesForProject,
+      projectId: projectId,
+    },
+  });
+
+  if (unitTypePickListError) throw unitTypePickListError;
+
+  return (
+    <Dialog open={open} onClose={handleClose} fullWidth maxWidth='sm'>
+      <DialogTitle>Create Units</DialogTitle>
+      <DialogContent sx={{ my: 2 }}>
+        {errorState.errors.length > 0 && (
+          <Box sx={{ mb: 2 }}>
+            <ErrorAlert key='errors' errors={errorState.errors} fixable />
+          </Box>
+        )}
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            <FormSelect
+              value={unitType ? { code: unitType } : null}
+              label={getRequiredLabel('Unit Type', false)}
+              placeholder='Select Unit Type'
+              loading={unitTypePickListLoading}
+              options={unitTypePickList || []}
+              onChange={(_event, option) => {
+                if (isPickListOption(option)) {
+                  setUnitType(option.code);
+                }
+              }}
+              maxWidth={400}
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <NumberInput
+              label={getRequiredLabel('Number of Units to Add', true)}
+              value={numberOfUnits ?? ''}
```

Let's add a limit (both here and in the backend) to the number of units you are allowed to bulk add? I'm able to get my dev environment to hang for some time by telling it to create 100000 units.

---

I'm also seeing a silent (swallowed) error by telling it to create 1000000000000000 units (not castable to int). Adding a `<ApolloErrorAlert error={errorState.apolloError} />` solves the swallowing, which I guess we should do anyway so that apollo errors are surfaced; but maybe this should be a validation error that's fixable by the user

---

thanks, added!